### DISCUSSION TITLE
Same loss every epoch with torch.fft in my code

### DISCUSSION THREAD

**Discussion by Sccrol** [Project-MONAI/MONAI#7831] [Q&A]
Hello everyone, I have been trying to reduce motion artifact on Liver Dynamic Contrast Enhanced DCE MRI. 
My idea is to deal with the k-space correction (mainly by correct the phase of the fourier transformations representation of the images, leave the amplitude alone). 
Here is my code:
```
for epoch in range(max_epochs):
    print("-" * 10)
    print(f"epoch {epoch + 1}/{max_epochs}")
    model.train()
    epoch_loss = 0
    step = 0
    for batch_data in train_loader:
        step += 1
        inputs, labels = (
            batch_data["image"].to(device),
            batch_data["label"].to(device),
        )
        optimizer.zero_grad()
        
        fftn = torch.fft.fftn(inputs,dim=(-3, -2), norm="ortho")
        fftn_c = torch.fft.fftshift(fftn, dim=(-3,-2))
        
        model_in = fftn_c.angle()
        model_out = model(model_in)
        
        bb_i = torch.sin(model_out) * fftn_c.abs()
        bb_r = torch.cos(model_out) * fftn_c.abs()
        bb = torch.complex(bb_r,bb_i)
        outputs = torch.abs(bb)       
        
        loss = loss_function(outputs, labels-inputs)  # + loss_function_3(outputs, labels-inputs)
        loss.backward()
        optimizer.step()
        epoch_loss += loss.item()
        print(
            f"{step}/{len(train_ds) // train_loader.batch_size}, "
            f"train_loss: {loss.item():.4f}")
    epoch_loss /= step
    epoch_loss_values.append(epoch_loss)
    print(f"epoch {epoch + 1} average loss: {epoch_loss:.4f}")
```
where _label_ is the clean image (ground truth) and _inputs_ is the image with simulated motions. The _model_ is monai.networks.nets.UNet, and the _loss_function_ is torch.nn.L1Loss().
However, the model does not train, gives the same loss in every epoch. 

Thanks for your replies, any help is very much appreciated.
